core/src/main/java/org/apache/carbondata/core/scan/filter/executer/ExcludeFilterExecuterImpl.java

```diff
@@ -43,6 +52,11 @@ public ExcludeFilterExecuterImpl(DimColumnResolvedFilterInfo dimColEvaluatorInfo
     this.segmentProperties = segmentProperties;
     FilterUtil.prepareKeysFromSurrogates(dimColEvaluatorInfo.getFilterValues(), segmentProperties,
         dimColEvaluatorInfo.getDimension(), dimColumnExecuterInfo);
+    // check if inverted index exists for the filter column
+    isInvertedIndexColumn = CarbonUtil
+        .hasEncoding(dimColEvaluatorInfo.getDimension().getEncoder(),
+            Encoding.INVERTED_INDEX);
+    isSortColumn = dimColEvaluatorInfo.getDimension().getColumnSchema().isSortColumn();
```

Comment:
During data load, according to the implementation, And operator is applied between isUseInvertedIndex and isSortColumn property. If anyone is false it will not sort the column vertically and will not create inverted index for that column. But in master schema the boolean flag will still be according to the values specified by the user. Therefor we need to check false as if anyone is false inverted index will not exist for that column